cli/disputer.go

```diff
@@ -238,6 +238,15 @@ var disputerStartCmd = &cli.Command{
 
 			// TODO: Parallelizeable
 			for _, dl := range dls {
+				// CHECK: if miner waller balance is zero then skip sending dispute message
+				walletBalance, err := api.WalletBalance(ctx, dl.miner)
+				if err != nil {
+					return xerrors.Errorf("failed to get wallet balance while checking to send dispute messages to miner %w: %w", dl.miner, err)
+				}
+				if walletBalance.Equals(big.Zero()) {
+					disputeLog.Warnw("wallet balance is zero, skipping dispute message", "wallet", dl.miner)
+					return nil
```

Comment:
This will skip processing the rest of the tipset. Instead, I think we need an early return of `nil, nil` inside "`makeDisputeWindowedPosts` if the miner has no balance.

Reply:
will add this change soon